plugin/src/commonObjects.ts

```diff
@@ -1,63 +1,141 @@
 export const globals = new Set([
-  'this',
-  'console',
-  'performance',
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects
+
+  // Value properties
+  'globalThis',
+  'Infinity',
+  'NaN',
+  'undefined',
+
+  // Function properties
+  'eval',
+  'isFinite',
+  'isNaN',
+  'parseFloat',
+  'parseInt',
+  'decodeURI',
+  'decodeURIComponent',
+  'encodeURI',
+  'encodeURIComponent',
+  'escape',
+  'unescape',
+
+  // Fundamental objects
+  'Object',
+  'Function',
+  'Boolean',
+  'Symbol',
+
+  // Error objects
+  'Error',
+  'AggregateError',
+  'EvalError',
+  'RangeError',
+  'ReferenceError',
+  'SyntaxError',
+  'TypeError',
+  'URIError',
+  'InternalError',
+
+  // Numbers and dates
+  'Number',
+  'BigInt',
+  'Math',
   'Date',
+
+  // Text processing
+  'String',
+  'RegExp',
+
+  // Indexed collections
   'Array',
-  'ArrayBuffer',
   'Int8Array',
-  'Int16Array',
-  'Int32Array',
   'Uint8Array',
   'Uint8ClampedArray',
+  'Int16Array',
   'Uint16Array',
+  'Int32Array',
   'Uint32Array',
+  'BigInt64Array',
+  'BigUint64Array',
   'Float32Array',
   'Float64Array',
-  'HermesInternal',
+
+  // Keyed collections
+  'Map',
+  'Set',
+  'WeakMap',
+  'WeakSet',
+
+  // Structured data
+  'ArrayBuffer',
+  'SharedArrayBuffer',
+  'DataView',
+  'Atomics',
   'JSON',
-  'Math',
-  'Number',
-  'Object',
-  'String',
-  'Symbol',
-  'undefined',
+
+  // Managing memory
+  'WeakRef',
+  'FinalizationRegistry',
+
+  // Control abstraction objects
+  'Iterator',
+  'AsyncIterator',
+  'Promise',
+  'GeneratorFunction',
+  'AsyncGeneratorFunction',
+  'Generator',
+  'AsyncGenerator',
+  'AsyncFunction',
+
+  // Reflection
+  'Reflect',
+  'Proxy',
+
+  // Internationalization
+  'Intl',
+  // 'Intl.Collator',
+  // 'Intl.DateTimeFormat',
+  // 'Intl.DisplayNames',
+  // 'Intl.DurationFormat',
+  // 'Intl.ListFormat',
+  // 'Intl.Locale',
+  // 'Intl.NumberFormat',
+  // 'Intl.PluralRules',
+  // 'Intl.RelativeTimeFormat',
+  // 'Intl.Segmenter',
+
+  // Other stuff
   'null',
-  'UIManager',
+  'this',
+  'global',
+  'console',
+  'performance',
+  'queueMicrotask',
   'requestAnimationFrame',
   'setImmediate',
-  'queueMicrotask',
+
+  // Hermes
+  'HermesInternal',
+
+  // Babel
+  'arguments', // from spreading `...args` (see PR #882)
+
+  // Reanimated
   '_WORKLET',
-  'arguments',
-  'Boolean',
-  'parseInt',
-  'parseFloat',
-  'Map',
-  'WeakMap',
-  'Proxy',
-  'WeakRef',
-  'Set',
-  'Intl',
   '_log',
   '_scheduleOnJS',
   '_makeShareableClone',
   '_updateDataSynchronously',
-  'eval',
   '_updatePropsPaper',
   '_updatePropsFabric',
   '_removeFromPropsRegistry',
-  'RegExp',
-  'Error',
-  '__ErrorUtils',
```

Comment:
Are you sure we can get rid of __ErrorUtils here?

Reply:
Yep, it's only used as `global.__ErrorUtils` so the plugin would capture only `global` but this is already allowlisted so it won't. I've tested it using one demo from WorkletExample.tsx which throws JS Errors on the UI runtime.